<?xml version="1.0"?>
<!-- Provided by sleutene for convenience -->
<launch>
  <!-- IPv4 address of your drone -->
  <arg name="ip" default="192.168.1.1" />
  <!-- Ultrasound frequency (7 or 8). -->
  <arg name="freq" default="8" />
  <!-- Launch ardrone_driver. -->
  <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver"
      output="screen" clear_params="true" args="-ip $(arg ip)">

    <!-- controller parameters -->
    <param name="pid_xy_kp" value="0.15" /> 
    <param name="pid_xy_ki" value="0.01" />
    <param name="pid_xy_kd" value="0.05" /> 
    <param name="pid_z_kp" value="1.5" />
    <param name="pid_z_ki" value="0.02" />
    <param name="pid_z_kd" value="0.2" />
    <param name="pid_yaw_kp" value="1.25" />
    <param name="pid_yaw_ki" value="0.01" />
    <param name="pid_yaw_kd" value="0.2" />

    <param name="outdoor" value="0" />
    <param name="max_bitrate" value="4000" />
    <param name="bitrate" value="4000" />
    <param name="navdata_demo" value="0" />
    <param name="flight_without_shell" value="0" />
    <param name="altitude_max" value="1600" />
    <param name="altitude_min" value="50" />
    <param name="euler_angle_max" value="0.21" />
    <param name="control_vz_max" value="700" />
    <param name="control_yaw" value="1.75" />
    <param name="detect_type" value="10" />
    <param name="enemy_colors" value="3" />
    <param name="detections_select_h" value="32" />
    <param name="detections_select_v_hsync" value="128" />
    <param name="enemy_without_shell" value="0" />
    <param name="ultrasound_freq" value="$(arg freq)" />
    <param name="realtime_navdata" value="true" />
    <param name="realtime_video" value="true" />
    <!-- Covariance Values (3x3 matrices reshaped to 1x9)-->
    <rosparam param="cov/imu_la">[0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 0.0, 0.0, 0.1]</rosparam>
    <rosparam param="cov/imu_av">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
    <rosparam param="cov/imu_or">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 100000.0]</rosparam>
  </node>
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ardrone_practicals)/config/rviz_vi_ekf.rviz"/>

  <!-- Launch estimator / control arp_node. -->
  <node name="arp_node" pkg="ardrone_practicals" type="arp_node" output="screen">
    <!-- Frontend parameters -->
    <param name="min_matches" value="5" />
    <param name="max_bow_results" value="50" />
    <!-- Camera parameters-->
    <param name="image_width" value="640" />
    <param name="image_height" value="360" />
    <param name="fu" value="582.055426" />
    <param name="fv" value="579.306311" />
    <param name="cu" value="316.003794" />
    <param name="cv" value="205.449078" />
    <param name="k1" value="-0.563630" />
    <param name="k2" value="0.361622" />
    <param name="p1" value="-0.011630" />
    <param name="p2" value="-0.001178" />
    <!-- BRISK detector parameters -->
    <param name="uniformity_radius" value="35" />
    <param name="octaves" value="5" />
    <param name="absolute_threshold" value="70" />
    <param name="max_num_kpt" value="1000" />
    <!-- We are not loading a map but we need the param for the startup to succeed -->
    <param name="map_camera_focal_length" value="390" />

    <param name="map" value="okvis2-slam-final_map_dronelab1.csv" />
    <param name="occupancymap" value="supereight2-densemap_dronelab1.dat" />
    <rosparam param="pointB">[1.0, 1.0, 1.0]</rosparam>
    <param name="inflation" value="2" />
    <param name="occ_thr" value="1" />

  
    <rosparam param="T_SC">[-0.00195087, -0.03257782,  0.99946730, 0.17409445, -0.99962338, -0.02729525, -0.00284087, 0.02255834, 0.02737326, -0.99909642, -0.03251230, 0.00174723, 0.00000000,  0.00000000,  0.00000000, 1.00000000]</rosparam>

  </node>

  

</launch>
